DOC-2266: Add new documentation for `ExportPDF` plugin. (#3078)
tinymce/tinymce-docs: modules/ROOT/pages/exportpdf.adoc

= Export to PDF plugin
:navtitle: Export to PDF

> :description: The Export to PDF feature lets you generate a PDF file directly from the editor.

Reviewer: Suggestion if we want to avoid using `you`
```suggestion
:description: The Export to PDF feature provides the ability to generate a PDF file directly from the editor.
```

:description_short: Generate a PDF file directly from the editor.
:keywords: plugin, exportpdf

> :pluginname: Export PDF

Author: ```suggestion
:pluginname: Export to PDF
```

:plugincode: exportpdf
:plugincategory: premium

include::partial$misc/admon-paid-addon-pricing.adoc[]

include::partial$misc/admon-requires-7.0v.adoc[]

> The Export to PDF functionality gathers the HTML produced using the `editor.getcontent()` method and the default editor content styles, along with the styles specified in the configuration. This data is transmitted to the {productname} Cloud Services HTML to PDF converter service. The service processes this information to create a PDF file, which is then sent back to the user's browser for saving to their local disk.

Reviewer: ```suggestion
The Export to PDF functionality gathers the HTML produced using the `editor.getcontent()` method and the default editor content styles, along with the styles specified in the configuration options for the plugin. This data is transmitted to the {productname} Cloud Services HTML to PDF converter service. The service processes this information to create a PDF file, which is then sent back to the user's browser for saving to their local disk.
```

== Interactive example

[NOTE]
.{pluginname} demo
====

> Demos containing the {pluginname} plugin are currently only available on our main website. Check out https://www.tiny.cloud/tinymce/features/exportpdf/[the {pluginname} demo], or the https://www.tiny.cloud/[demo on our home page], to try it out today.

Author: ```suggestion
Demos containing the {pluginname} plugin are currently only available on our main website. Check out https://www.tiny.cloud/tinymce/features/export-pdf/[the {pluginname} demo], or the https://www.tiny.cloud/[demo on our home page], to try it out today.
```

====

== Basic setup

To add the {pluginname} plugin to the editor, add `{plugincode}` to the `plugins` option in the editor configuration.

For example:

[source,js]
----
tinymce.init({
  selector: 'textarea',
  plugins: 'exportpdf',
  toolbar: 'exportpdf',
  exportpdf_service_url: '<service url>'
});
----

== Options

The following configuration options affect the behavior of the {pluginname} plugin.

> include::partial$configuration/{plugincode}.adoc[][leveloffset=+1]

Reviewer 2: ```suggestion
include::partial$configuration/{plugincode}.adoc[leveloffset=+1]
```
fix heading levels in the navbar
before:
<img width="304" alt="image" src="https://github.com/tinymce/tinymce-docs/assets/40410898/8964fd15-47e6-405a-a603-53e57f53298b">

after:
<img width="271" alt="image" src="https://github.com/tinymce/tinymce-docs/assets/40410898/1710e48e-648b-4933-ae2f-19f62d0bb491">

= Export to PDF plugin
:navtitle: Export to PDF
:description: The Export to PDF feature lets you generate a PDF file directly from the editor.
:description_short: Generate a PDF file directly from the editor.
:keywords: plugin, exportpdf, export to pdf
:pluginname: Export to PDF

> :pluginminimumplan: <tier>

Author: ```suggestion
```

= Export to PDF plugin
:navtitle: Export to PDF
:description: The Export to PDF feature lets you generate a PDF file directly from the editor.
:description_short: Generate a PDF file directly from the editor.
:keywords: plugin, exportpdf, export to pdf
:pluginname: Export to PDF
// values for <tier>
// opensource plugin = tieroneplan

> // premium plugin    = tiertwoplan or tierthreeplan or enterpriseplan

Author: @ShiridiGandham Can you confirm what we want to set this to.

Reviewer 3: The approved Export to word plugin docs also has this open.

Author: ```suggestion
```

== Interactive example

liveDemo::{plugincode}[]

== Basic setup

To add the {pluginname} plugin to the editor, add `{plugincode}` to the `plugins` option in the editor configuration.

For example:

[source,js]
----
tinymce.init({
  selector: 'textarea',
  plugins: 'exportpdf',
  toolbar: 'exportpdf',
});
----

== Options

The following configuration options affect the behavior of the {pluginname} plugin.

include::partial$configuration/{plugincode}.adoc[][leveloffset=+1]

include::partial$misc/plugin-toolbar-button-id-boilerplate.adoc[]

include::partial$misc/plugin-menu-item-id-boilerplate.adoc[]

== Commands

The {pluginname} plugin provides the following {productname} commands.

include::partial$commands/{plugincode}-cmds.adoc[]

== Events

The {pluginname} plugin provides the following events.

> include::partial$events/{plugincode}-events.adoc[]

Reviewer 4: We have some events intended for Schifer tracking but not for public usage. So, yes you can remove this Event section.

> == Events
> 
> The {pluginname} plugin provides the following events.
> 
> include::partial$events/{plugincode}-events.adoc[]

Author: ```suggestion
```

== Interactive example

liveDemo::{plugincode}[]

== Basic setup

To add the {pluginname} plugin to the editor, add `{plugincode}` to the `plugins` option in the editor configuration.

For example:

[source,js]
----
tinymce.init({
  selector: 'textarea',
  plugins: 'exportpdf',
  toolbar: 'exportpdf',
});
----

== Options

The following configuration options affect the behavior of the {pluginname} plugin.

include::partial$configuration/{plugincode}.adoc[][leveloffset=+1]

include::partial$misc/plugin-toolbar-button-id-boilerplate.adoc[]

include::partial$misc/plugin-menu-item-id-boilerplate.adoc[]

== Commands

The {pluginname} plugin provides the following {productname} commands.

include::partial$commands/{plugincode}-cmds.adoc[]

== Events

The {pluginname} plugin provides the following events.

include::partial$events/{plugincode}-events.adoc[]

> ////

Author: ```suggestion
```

== Interactive example

liveDemo::{plugincode}[]

== Basic setup

To add the {pluginname} plugin to the editor, add `{plugincode}` to the `plugins` option in the editor configuration.

For example:

[source,js]
----
tinymce.init({
  selector: 'textarea',
  plugins: 'exportpdf',
  toolbar: 'exportpdf',
});
----

== Options

The following configuration options affect the behavior of the {pluginname} plugin.

include::partial$configuration/{plugincode}.adoc[][leveloffset=+1]

include::partial$misc/plugin-toolbar-button-id-boilerplate.adoc[]

include::partial$misc/plugin-menu-item-id-boilerplate.adoc[]

== Commands

The {pluginname} plugin provides the following {productname} commands.

include::partial$commands/{plugincode}-cmds.adoc[]

[cols="1,3",options="header"]
|===
|Command         |Description
|exportpdf_service_url   |triggers a function that converts the content of the editor into a PDF format using a specified or default converter URL.
|===

.Examples
[source,js]
----
tinymce.activeEditor.execCommand('exportpdf');
----

> ////

Reviewer 4: This is something to remove

Author: ```suggestion
```
